sigstore/_internal/tuf.py

```diff
@@ -0,0 +1,160 @@
+
+    return (data_dir / tuf_base), (cache_dir / tuf_base)
+
+
+class TrustUpdater:
+    """Internal trust root (certificates and keys) downloader.
+
+    TrustUpdater discovers the currently valid certificates and keys and
+    securely downloads them from the remote TUF repository at 'url'.
+
+    TrustUpdater expects to find an initial root.json in either the local
+    metadata directory for this URL, or (as special case for the sigstore.dev
+    production and staging instances) in the application resources.
+    """
+
+    def __init__(self, url: str) -> None:
+        self._repo_url = url
+        self._updater: Optional[Updater] = None
+
+        self._metadata_dir, self._targets_dir = _get_dirs(url)
+
+        # intialize metadata dir
+        tuf_root = self._metadata_dir / "root.json"
+        if not tuf_root.exists():
+            if self._repo_url == DEFAULT_TUF_URL:
+                fname = "root.json"
+            elif self._repo_url == STAGING_TUF_URL:
+                fname = "staging-root.json"
+            else:
+                raise Exception(f"TUF root not found in {tuf_root}")
+
+            self._metadata_dir.mkdir(parents=True, exist_ok=True)
+            root_json = read_embedded(fname)
+            with tuf_root.open("wb") as io:
+                io.write(root_json)
+
+        # Initialize targets cache dir
+        # NOTE: Could prime the cache here with any embedded certs/keys
+        self._targets_dir.mkdir(parents=True, exist_ok=True)
+
+        logger.debug(f"TUF metadata: {self._metadata_dir}")
+        logger.debug(f"TUF targets cache: {self._targets_dir}")
+
+    @classmethod
+    def production(cls) -> "TrustUpdater":
+        return cls(DEFAULT_TUF_URL)
+
+    @classmethod
+    def staging(cls) -> "TrustUpdater":
+        return cls(STAGING_TUF_URL)
+
+    def _setup(self) -> Updater:
+        """Initialize and update the toplevel TUF metadata"""
+        updater = Updater(
+            metadata_dir=str(self._metadata_dir),
+            metadata_base_url=f"{self._repo_url}",
+            target_base_url=f"{self._repo_url}targets/",
+            target_dir=str(self._targets_dir),
+            fetcher=_fetcher,
+        )
+
+        # NOTE: we would like to avoid refresh if the toplevel metadata is valid.
+        # https://github.com/theupdateframework/python-tuf/issues/2225
+        updater.refresh()
+        return updater
+
+    def _get(self, usage: str) -> List[bytes]:
+        """Return all active targets with given usage"""
+        if not self._updater:
+            self._updater = self._setup()
+
+        data = []
+
+        # NOTE: _updater has been fully initialized at this point, but mypy can't see that.
+        targets = self._updater._trusted_set.targets.signed.targets  # type: ignore[union-attr]
+        for target_info in targets.values():
+            custom = target_info.unrecognized_fields["custom"]["sigstore"]
+            if custom["status"] == "Active" and custom["usage"] == usage:
```

Comment:
note that on the verification path, Expired/Inactive targets can still be used to verify against. It might not be worth filtering these out, but rather returning a List of bytes that has this metadata attached

Reply:
thanks -- I might add an argument like `active_only: bool` to the getters instead but this is a good point, not sure how I missed it.

I'll file a new issue on these, but maybe tomorrow

Reply:
following up in #375